tests/checkers/unittest_imports.py

```diff
@@ -41,42 +41,57 @@ def test_relative_beyond_top_level(self) -> None:
             self.checker.visit_importfrom(module.body[2].body[0])
 
     @staticmethod
-    @pytest.mark.xfail(
-        reason="epylint manipulates cwd; these tests should not be using epylint"
-    )
-    def test_relative_beyond_top_level_two() -> None:
-        output, errors = lint.py_run(
-            f"{os.path.join(REGR_DATA, 'beyond_top_two')} -d all -e relative-beyond-top-level",
-            return_std=True,
+    def test_relative_beyond_top_level_two(capsys: CaptureFixture[str]) -> None:
+        Run(
+            [
+                f"{os.path.join(REGR_DATA, 'beyond_top_two')}",
+                "-d all",
+                "-e relative-beyond-top-level",
+            ],
+            exit=False,
         )
+        output, errors = capsys.readouterr()
+
         top_level_function = os.path.join(
             REGR_DATA, "beyond_top_two/namespace_package/top_level_function.py"
         )
-        output2, errors2 = lint.py_run(
-            f"{top_level_function} -d all -e relative-beyond-top-level",
-            return_std=True,
+        Run(
+            [top_level_function, "-d all", "-e relative-beyond-top-level"],
+            exit=False,
         )
+        output2, errors2 = capsys.readouterr()
 
-        assert len(output.readlines()) == len(output2.readlines())
-        assert errors.readlines() == errors2.readlines()
+        # The first package fails to lint
+        assert len(output.split("\n")) == 1
```

Comment:
Should we leave this as 5 and `xfail` it? Tests are documentation, and I don't want to document the wrong result.

Reply:
Well, maybe the result is correct if the expectation is that you have to `cd` to a namespace package you want to lint.

Reply:
Yeah I was meaning to ask this. Is the new behavior how we want to do this? Or is this a temporary result because of improved but not perfect namespace package handling?

If this is desired (ie., you can't lint a package like this) I think it makes sense to just have this as a test itself.

Reply:
Thanks for giving me time to double-check. Yes, you need to `cd` or otherwise do something fancy (with a hook?) if you want to find namespace packages. This matches the behavior of python's `import` statement, so far as I can tell. So the result is correct.

To go back to the original issue linked in these tests, we're just trying to test for the absence of `no-name-in-module` or `relative-beyond-top-level`.

Reply:
I was wrong about this after all. I guess the `beyond_top_two` _is_ on `sys.path` somewhere, so it should be linted. (I guess because `tests` is on sys.path?) See #7304.

Ultimately new tests should create their own isolated packages with code, to make this easier to reason about!